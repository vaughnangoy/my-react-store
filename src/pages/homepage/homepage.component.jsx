import React from 'react';

import './homepage.styles.scss';
import { DirectoryMenu } from "../../components/directory-menu/directory-menu.component";

export const HomePage = () => {
    return (
        <div className="homepage">
            <DirectoryMenu />
        </div>
    );
}