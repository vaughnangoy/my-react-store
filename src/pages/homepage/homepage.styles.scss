body {
    margin: 60px;
    padding: 20px;
}
.homepage {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 20px 80px;
}
