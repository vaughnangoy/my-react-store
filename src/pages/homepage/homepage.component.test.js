import React from 'react';
import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import { createStore } from 'redux';

import { HomePage } from './homepage.component';

const generateComponent = props => {
    const history = createBrowserHistory();
    const store = createStore(() => {}, {});
    return (
        <Provider store={store}>
            <Router history={history}>
                <>
                    <HomePage {...props} />
                </>
            </Router>
        </Provider>
    );
};

describe('HomePage', () => {
    afterEach(cleanup);
    it('renders', () => {
        const { asFragment } = render(generateComponent());
        expect(asFragment()).toMatchSnapshot();
    });

    it('should contain the directory component in document', () => {
        const { getByTestId } = render(generateComponent());
        expect(getByTestId('directory-menu')).toBeInTheDocument();
    });
});
