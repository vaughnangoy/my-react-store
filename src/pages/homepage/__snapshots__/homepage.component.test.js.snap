// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders 1`] = `
<DocumentFragment>
  <div
    class="homepage"
  >
    <div
      class="directory-menu"
      data-testid="directory-menu"
    >
      <div
        class="menu-item undefined"
      >
        <div
          class="bg"
          style="background-image: url(https://i.ibb.co/cvpntL1/hats.png);"
        />
        <div
          class="content"
        >
          <h1
            class="title"
          >
            HATS
          </h1>
          <span
            class="subtitle"
          >
            Shop Now
          </span>
        </div>
      </div>
      <div
        class="menu-item undefined"
      >
        <div
          class="bg"
          style="background-image: url(https://i.ibb.co/px2tCc3/jackets.png);"
        />
        <div
          class="content"
        >
          <h1
            class="title"
          >
            JACKETS
          </h1>
          <span
            class="subtitle"
          >
            Shop Now
          </span>
        </div>
      </div>
      <div
        class="menu-item undefined"
      >
        <div
          class="bg"
          style="background-image: url(https://i.ibb.co/0jqHpnp/sneakers.png);"
        />
        <div
          class="content"
        >
          <h1
            class="title"
          >
            SNEAKERS
          </h1>
          <span
            class="subtitle"
          >
            Shop Now
          </span>
        </div>
      </div>
      <div
        class="menu-item large"
      >
        <div
          class="bg"
          style="background-image: url(https://i.ibb.co/GCCdy8t/womens.png);"
        />
        <div
          class="content"
        >
          <h1
            class="title"
          >
            WOMENS
          </h1>
          <span
            class="subtitle"
          >
            Shop Now
          </span>
        </div>
      </div>
      <div
        class="menu-item large"
      >
        <div
          class="bg"
          style="background-image: url(https://i.ibb.co/R70vBrQ/men.png);"
        />
        <div
          class="content"
        >
          <h1
            class="title"
          >
            MENS
          </h1>
          <span
            class="subtitle"
          >
            Shop Now
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
